modules/core/src/main/java/org/apache/ignite/internal/manager/IgniteComponent.java

```diff
@@ -17,15 +17,19 @@
 
 package org.apache.ignite.internal.manager;
 
+import java.util.concurrent.CompletableFuture;
+
 /**
  * Common interface for ignite components that provides entry points for component lifecycle flow.
  */
 public interface IgniteComponent {
     /**
      * Starts the component. Depending on component flow both configuration properties listeners, meta storage watch registration, starting
      * thread pools and threads goes here.
+     *
+     * @return Future that will be completed when the asynchronous part of the start is processed.
      */
-    void start();
+    CompletableFuture<Void> start();
 
```

Comment:
Right after the merge, please broadcast in ai3 slack channel that you've introduced an engine that allows to have an async components start and that no *new* async logic was added within your PR.